webapp/app/components/userMenu/userMenu.component.js

```diff
@@ -0,0 +1,62 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Map } from 'immutable';
+import IconButton from 'material-ui/IconButton';
+import Menu, { MenuItem } from 'material-ui/Menu';
+import { FormattedMessage } from 'react-intl';
+
+import { UserAvatar } from '../userAvatar/userAvatar.component';
+import messages from './userMenu.messages';
+
+
+export class UserMenu extends PureComponent {
+  static propTypes = {
+    user: PropTypes.instanceOf(Map),
+    onSignOut: PropTypes.func.isRequired,
+  };
+
+  state = {
+    anchorEl: null,
+  };
+
+  get open() {
+    return !!this.state.anchorEl;
+  }
+
+  handleClose = () => {
+    this.setState({ anchorEl: null });
+  };
+
+  handleMenu = ({ currentTarget: anchorEl }) => this.setState({ anchorEl });
+
+  render = () => (
+    <div>
+      <IconButton
+        aria-owns={this.open ? 'menu-appBar' : null}
+        aria-haspopup="true"
+        onClick={this.handleMenu}
```

Comment:
Pierdołka: skoro funkcja do zamykania nazywa się `handleClose` to tutaj mogło by być `handleOpen` albo `handleClick`, bo `handleMenu` nie do końca mówi co to robi.